### PR TITLE
feat: support packageManager property in package.json

### LINKED ISSUE #10714: Use `packageManager` field in `package.json` for package manager detection
Currently NodeBB uses either `config.json` or it falls back to basic lockfile detection.
However, with the addition of `corepack`, [`package.json` can now carry that information in `packageManager` field](https://nodejs.org/dist/latest-v16.x/docs/api/all.html#all_packages_packagemanager).

I think it would be reasonable to also respect that field, though I'm not sure if it should take precedence over lockfiles (I guess it should be below explicit NodeBB configuration though).

Additionally, I think the logic for selecting package manager from package.json and lockfiles could be expanded to work without nconf.

I can try doing both when I know the preferred priority of packageManager/lockfile check :)

### PR DESCRIPTION
**Pull request by oplik0**
resolves #10714

I think `nconf.get('package_manager'` actually isn't used anywhere outside of `package-install`, so it shouldn't be an issue that using `packageManager` in `package.json` doesn't set that property.

Also, should there perhaps be some mention of how to use alternative package managers in docs?

Edit: changed the base to develop

### CODE REVIEW

File: src/cli/package-install.js [hunk 1]
```diff
@@ -70,42 +78,30 @@ pkgInstall.getPackageManager = () => {
...
 
+function getPackageManagerByLockfile() {
+	for (const [packageManager, lockfile] of Object.entries({ npm: 'package-lock.json', yarn: 'yarn.lock', pnpm: 'pnpm-lock.yaml' })) {
+		try {
+			fs.accessSync(path.resolve(__dirname, `../../${lockfile}`), fs.constants.R_OK);
+			return packageManager;
+		} catch (e) {}
+	}
+}
+
```
Comment (julianlam) [inline src/cli/package-install.js:104]: Look at you, refactoring my verbose code :smirk:

**Review by julianlam — APPROVED** [review]

**Comment by julianlam** [comment]
What is `corepack` :thinking:

**Comment by oplik0** [comment]
It's an experimental package manager manager for Node, included first with Node 16 (though it was also backported to 14).
It allows one to manage versions of npm, yarn and pnpm (currently), including configuring them within `package.json`.
It's quite nice, just running `corpack enable` once basically gets you `yarn`/`pnpm` as needed :)
https://nodejs.org/api/corepack.html

**Comment by julianlam** [comment]
Added new config to docs

**Comment by oplik0** [comment]
This was actually already in NodeBB though, so the note about availability doesn't seem right. If anything the new code actually overrides it in certain situation (`packageManager` field in `package.json` takes precedence. This has to be set manually and is consistent with tooling, since presumably if one sets it there they then ran that specific version of the tool via corepack)

**Comment by julianlam** [comment]
ah! Good to know, I'll remove that availability line :smile:

**Comment by julianlam** [comment]
Hang on, `package_manager` or `packageManager`? If `package_manager` wasn't around before, then it is not available yet, no?

**Comment by oplik0** [comment]
`package_manager` in `config.json` has been around for quite some time. As you mentioned in the issue (https://github.com/NodeBB/NodeBB/issues/10714), now than NodeJS is bringing native package manager management it can be considered the legacy method.

The new method is using `package.json` instead and setting the `packageManager` field there

**Pull request merged**